ramls/request-move.raml

```diff
@@ -0,0 +1,39 @@
+#%RAML 1.0
```

Comment:
This was a request to avoid introducing a version change to the existing circulation.raml.

Reply:
I asked for this to be split out.

The first version of this work used the request as the incoming body for this endpoint. I hadn't had the chance to express technical requirements in any detail during this work :-(

I wasn't keen on that, as it allows any manipulation of the request during a move, which at the very least extends the surface area.

A counter argument is if there is a desire to change multiple parts of the request during a move then this would be a way to let the client do so without further changes.

Given we didn't have a good way of evaluating that future, I suggested being defensive and splitting this out into an experimental interface.

I'm getting more and more cautious of prematurely committing to new API that then require breaking changes, and this is a trick I'm using to allow work to move on without the need for technical design conversations holding it up (as folks have raised concerns about that).

I'm aware that my views on recent API design are likely to be unpopular.

Reply:
I thought that is what was going on, but I guess my surprise was that there was not a new interface for the move endpoint in the module descriptor.

Reply:
Oh, there should be, it needs to be in the module descriptor to be used by the UI modules in the snapshot environments

Reply:
So... what is desired? Should I create a new interface in the module descriptor? Then shouldn't it have a new Vertex route and resource? I am not understanding the reasoning for seperating out an operation of a resource item into a new interface.

Reply:
The reasoning behind the separation is to protect the rest of the `circulation` interface from compatibility breaking changes needed for `move`.

You and I discussed the tradeoffs between having a few properties vs. the request schema. This is my way of trying to allow us to change our minds without lots of coordination effort.

Does that make sense?

Reply:
@wwelling You've already split the endpoint into a separate interface in RAML. For Okapi to understand that the endpoint is in a separate interface, it needs to also be reflected in the module descriptor.

(The endpoints described in the RAML should match the endpoints bound in the module descriptor `provides` array)

Does that make sense?

Reply:
The implementation is unaffected by these changes, as the endpoint itself remains the same, it is only logically part of a different Okapi interface.

Reply:
All that really needs to be done is to break out the move endpoint in `circulation` into a new provided interface that just has that endpoint in the module descriptor. This is similar to `requests-reports`.

Reply:
ok, will do

Reply:
done